custom_components/unifi_access/hub.py

```diff
@@ -212,16 +213,33 @@ def on_message(self, ws: websocket.WebSocketApp, message):
                             door_id,
                         )
                 case "access.data.device.update":
-                    door_id = update["data"]["door"]["unique_id"]
-                    _LOGGER.info("Device Update via websocket %s", door_id)
-                    if door_id in self.doors:
-                        existing_door = self.doors[door_id]
-                        self.update_door(door_id)
-                        _LOGGER.info(
-                            "Door name %s with ID %s updated",
-                            existing_door.name,
-                            door_id,
-                        )
+                    device_type = update["data"]["device_type"]
+                    if device_type == "UAH":
+                        door_id = update["data"]["door"]["unique_id"]
+                        _LOGGER.info("Device Update via websocket %s", door_id)
+                        if door_id in self.doors:
+                            existing_door = self.doors[door_id]
+                            self.update_door(door_id)
+                            _LOGGER.info(
+                                "Door name %s with ID %s updated",
+                                existing_door.name,
+                                door_id,
+                            )
+                    elif device_type == "UAH-Ent":
+                        for door_object in update["data"]["extensions"]:
+                            door_id = door_object["target_value"]
+                            _LOGGER.info("Device Update via websocket %s", door_id)
+                            if door_id in self.doors:
+                                existing_door = self.doors[door_id]
+                                self.update_door(door_id)
+                                _LOGGER.info(
+                                    "Door name %s with ID %s updated",
+                                    existing_door.name,
+                                    door_id,
+                                )
+                                asyncio.run_coroutine_threadsafe(
+                                    existing_door.publish_updates(), self.loop
```

Comment:
I'm still new to the events in HA, but publishing updates only at the end of this method causes only Door 8 not to stuck. Am I missing something?

Reply:
> only Door 8 not to stuck.

What do you mean by this?

Either way, we'd want to publish updates and trigger events **after** processing the message.

I don't have a UAH enterprise but perhaps you can share the message format and we can incorporate it here.